schulcloud/CoreDataObserver.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  CoreDataObserver.swift
+//  schulcloud
+//
+//  Created by Max Bothe on 20.02.18.
+//  Copyright © 2018 Hasso-Plattner-Institut. All rights reserved.
+//
+
+import CoreData
+
+class CoreDataObserver {
+
+    static let shared = CoreDataObserver()
+
+    func startObserving() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(managedObjectContextObjectsDidChange),
+                                               name: NSNotification.Name.NSManagedObjectContextObjectsDidChange,
+                                               object: CoreDataHelper.viewContext)
+    }
+
+    func stopObserving() {
+        NotificationCenter.default.removeObserver(self,
+                                                  name: NSNotification.Name.NSManagedObjectContextObjectsDidChange,
+                                                  object: CoreDataHelper.viewContext)
+    }
+
+    @objc func managedObjectContextObjectsDidChange(notification: NSNotification) {
```

Comment:
I'm torn with this function. On one hand, I like the fact that we have one place where object changed notification is handled instead of being scattered all over the place, on the other hand it feels a bit odd because there are a lot of repetitive sequence.
What do you think of having this class registering block for every `startObserving(work: ([String: Any] -> Void) -> NSObjectProtocol` and `stopObserving(observer: NSObjectProtocol)`, and each piece of code that needs to observe just submit work to be done? This is a naive question, I would like to know your opinion on that alternative.